src/oop/special_class_method/hard/q9/MainTest.java

```diff
@@ -0,0 +1,216 @@
+        assertTrue(testClass.hasField(intFieldName, int.class),
+                "Your " + className + " class is missing the " + intFieldName + " attributes.");
+        assertTrue(testClass.hasModifier(intFieldName, "private"),
+                "Your " + intFieldName + " does not have the correct visibility modifier.");
+    }
+
+    private static Stream<Arguments> constructorInputProvider() {
+        return Stream.of(
+                Arguments.of(4535.4, 300),
+                Arguments.of(234.4, 100),
+                Arguments.of(654.45, 365),
+                Arguments.of(43.3, 14)
+
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("constructorInputProvider")
+    public void earthClassHasCorrectConstructor(double value1, int value2) throws Throwable {
+        Object[][] arguments = {
+                {value1, double.class},
+                {value2, int.class}
+        };
+        Class<?>[] constructorClasses = {double.class, int.class};
+        Object classInstance = testClass.createInstance(arguments);
+        String incorrectDoubleValueMessage =
+                "Your " + className + " constructor does not initialize the " + doubleFieldName + " attribute to the correct value.";
+        String incorrectStringValueMessage =
+                "Your " + className + " constructor does not initialize the " + intFieldName + " attribute to the correct value.";
+        String incorrectConstructorVisibilityModifier =
+                "Your " + className + " constructor does not have the correct visibility modifier.";
+        _assertEquals(value1, testClass.getFieldValue(classInstance, doubleFieldName),
+                incorrectDoubleValueMessage);
+        _assertEquals(value2, testClass.getFieldValue(classInstance, intFieldName),
+                incorrectStringValueMessage);
+        assertTrue(testClass.hasModifier(constructorClasses, "public"), incorrectConstructorVisibilityModifier);
+    }
+
+    private static Stream<Arguments> doubleInputProvider() {
+        return Stream.of(
+                Arguments.of(4535.4, 432.2),
+                Arguments.of(234.4, 655.3),
+                Arguments.of(654.45, 453.3),
+                Arguments.of(43.3, 654.2)
+
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("doubleInputProvider")
+    public void earthClassHasCorrectSetLandMassMethod(double initialValue, double setValue) throws Throwable {
+        Object[][] arguments = {
+                {initialValue, double.class},
+                {0, int.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        Object[][] setSizeArguments = {
+                {setValue, double.class}
+        };
+        String[] setMethodModifiers = {"public"};
+        String incorrectSetMethodMessage = String.join(" ",
+                "Your", setDoubleMethodName, "does not correctly set the value of the", doubleFieldName, "attribute.");
```

Comment:
in this message and any other make sure that after a method name there is "method"